### ISSUE TITLE
Revise SIF export

### ISSUE THREAD

**Issue by cmungall** [geneontology/noctua#604]
background: SIF doesn't have any semantics, it's just a very simple graph format for Cytoscape. See http://manual.cytoscape.org/en/stable/Supported_Network_File_Formats.html#sif-format

Looking at the SIF export on http://geneontology.org/cam

What is the use case for SIF format? Presumably to visualize in Cytoscape, but what's the user story here? 

Currently exports look like this:

```
ether lipid biosynthetic process        occurs_in       monocyte
fatty acid synthase activity    part_of ether lipid biosynthetic process
fatty acid synthase activity_2  causally_upstream_of_or_within  establishment of endothelial intestinal barrier
fatty acid synthase activity_3  enabled_by      Fasn Mmus
fatty acid synthase activity_4  part_of neutrophil differentiation
ether lipid biosynthetic process_2      part_of neutrophil differentiation
cell development        has_output      goblet cell
tissue development      has_output      white adipose tissue
ether lipid biosynthetic process        part_of monocyte differentiation
cell development        occurs_in       intestine
fatty acid synthase activity    enabled_by      Fasn Mmus_2
fatty acid synthase activity    part_of monocyte differentiation
fatty acid synthase activity_4  enabled_by      Fasn Mmus_3
acylglycerone-phosphate reductase activity      part_of neutrophil differentiation
acylglycerone-phosphate reductase activity      part_of ether lipid biosynthetic process_2
acylglycerone-phosphate reductase activity      enabled_by      Dhrs7b Mmus
neutrophil differentiation      causally_upstream_of    inflammatory response
fatty acid synthase activity_2  causally_upstream_of_or_within  cell development
fatty acid synthase activity_4  part_of ether lipid biosynthetic process_2
ether lipid biosynthetic process_2      occurs_in       neutrophil
fatty acid synthase activity_2  enabled_by      Fasn Mmus_4
fatty acid synthase activity_3  causally_upstream_of_or_within  tissue development
monocyte differentiation        occurs_in       bone marrow
neutrophil differentiation      occurs_in       bone marrow_2
cell development        part_of establishment of endothelial intestinal barrier
```

This looks like a serialization of the expanded via in Noctua, i.e. export every instance and label it with the class label, but also uniquifying.

Not clear there is an advantage to seeing the whole set of abox triples this way in Cytoscape? Surely better to view this in Noctua where there is nesting etc.

I suggest instead the export is an activity-centered-entity-labeled view. i.e.

```
1. select all activity nodes A and activity-activity causal edges E
2. create a graph G = A,E
3. label nodes in A by following enabled_by/label
```

**Comment by thomaspd** [comment]
I agree with your suggestion-- I made the same suggestion earlier but didn't put it so elegantly! :)

**Comment by goodb** [comment]
For cytoscape purposes, I think it would be valuable to export the nodes
using GO accession numbers and uniprot identifiers as the keys.  I guess
this isn't a completely plain SIF export - maybe another option: "export
for cytoscape".  If you do this, then you get easy access to a lot of other
data via Cytoscape plugins (e.g. http://apps.cytoscape.org/apps/stringapp
).  See http://manual.cytoscape.org/en/stable/Node_and_Edge_Column_Data.html

**Comment by lpalbou** [comment]
@cmungall correct, the current downloadable SIF export is more or less a 1:1 mapping of the GO-CAMs we have (most relations are indeed included). 

I started to work on a more activity-centric export (https://github.com/geneontology/gocam-sif-pyexport#gene-product---gene-product-relationships) ~2 months ago but had other projects meanwhile and didn't had time to refine it until now.

@thomaspd one problem we have (e.g. my mockup presentation of activity-centric views) is very few GO-CAMs currently follow the GO-CAM model (e.g. very few have causal relationships, most are dependent of a BP node treaded as an activity node, etc) and I was searching for a solution to still represent those current models. I can create a strict activity-centric representation but my guess is we'll have less than 100 models with more than 1 node.

@goodb I agree but in the SIF format, you can not have a node described by both a label and an identifier, that's why I added a label parameter in the SIF generation process (https://github.com/geneontology/gocam-sif-pyexport#curies-vs-labels) to produce both labeled version (for biologists) and id version (for bioinformatician). I could give a direct access to both files on the GO-CAM site. Would that be ok ?

For reference, a test version of the CytoJs export of those SIF are here: http://cytocam.geneontology.cloud

Larger question: if the main target of this export is Cytoscape user, we could use a richer format

**Comment by vanaukenk** [comment]
@cmungall - should we keep this ticket open?
